utils/directories.cc

```diff
@@ -149,4 +151,108 @@ future<> directories::verify_owner_and_mode(fs::path path, recursive recursive)
     return do_verify_owner_and_mode(std::move(path), recursive, 0);
 }
 
+void directories::set_work_dir(const seastar::sstring& work_dir) {
+    _work_dir = fs::path{work_dir};
+}
+
+void directories::set_commitlog_dir(const seastar::sstring& commitlog_dir) {
+    _commitlog_dir = fs::path{commitlog_dir};
+}
+
+void directories::set_schema_commitlog_dir(const seastar::sstring& schema_commitlog_dir) {
+    _schema_commitlog_dir = fs::path{schema_commitlog_dir};
+}
+
+void directories::set_hints_dir(const seastar::sstring& hints_dir) {
+    _hints_dir = fs::path{hints_dir};
+}
+
+void directories::set_view_hints_dir(const seastar::sstring& view_hints_dir) {
+    _view_hints_dir = fs::path{view_hints_dir};
+}
+
+void directories::set_saved_caches_dir(const seastar::sstring& saved_caches_dir) {
+    _saved_caches_dir = fs::path{saved_caches_dir};
+}
+
+void directories::set_data_file_dirs(const std::vector<seastar::sstring>& data_file_dirs) {
+    std::vector<fs::path> dirs_paths;
+    dirs_paths.reserve(data_file_dirs.size());
+
+    for (const auto& dir_str : data_file_dirs) {
+        dirs_paths.emplace_back(dir_str);
+    }
+
+    _data_file_dirs = std::move(dirs_paths);
+}
+
+static seastar::sstring to_sstring(const fs::path& p) {
+    return seastar::sstring{p.c_str()};
+}
+
+seastar::sstring directories::get_commitlog_dir() const {
+    return to_sstring(_commitlog_dir);
+}
+
+seastar::sstring directories::get_schema_commitlog_dir() const {
+    return to_sstring(_schema_commitlog_dir);
+}
+
+seastar::sstring directories::get_hints_dir() const {
+    return to_sstring(_hints_dir);
+}
+
+seastar::sstring directories::get_view_hints_dir() const {
+    return to_sstring(_view_hints_dir);
+}
+
+seastar::sstring directories::get_saved_caches_dir() const {
+    return to_sstring(_saved_caches_dir);
+}
+
+std::vector<seastar::sstring> directories::get_data_file_dirs() const {
```

Comment:
I think that it's enough if this method returns const reference directly on _data_file_dirs. If caller is OK with using const reference, it will avoid copying of this vector, if not, it will just default-copy-construct it

Reply:
Right now the paths are stored as member variables (`fs::path` type). Should we change them to `sstring`?

The issue contains the information that storing paths as `fs::path` should be considered. That's why I did it this way. But maybe sstring is sufficient.

Comment:
https://github.com/scylladb/scylladb/issues/5626#issuecomment-904695991